packages/react-ui-core/src/List/List.js

```diff
@@ -0,0 +1,49 @@
+import React, { PureComponent } from 'react'
+import PropTypes from 'prop-types'
+import cn from 'classnames'
+import ListItem from './ListItem'
+
+export default class List extends PureComponent {
+  static propTypes = {
+    theme: PropTypes.object,
+    className: PropTypes.string,
+    nodeType: PropTypes.string,
+    Item: PropTypes.node,
+    items: PropTypes.array,
+    listItemNodeType: PropTypes.string,
+  }
+
+  static defaultProps = {
+    theme: {},
+    nodeType: 'ul',
+    listItemNodeType: 'li',
+    Item: ListItem,
+    items: [],
+  }
+
+  render() {
+    const {
+      theme,
+      className,
+      Item,
+      items,
+      listItemNodeType,
+      ...props
+    } = this.props
+
+    return (
+      <this.props.nodeType
+        className={cn(
+          theme.List,
+          className,
+        )}
+      >
+        {items.map((item, i) => (
+          <Item key={i} listItemNodeType={listItemNodeType} {...props}>
```

Comment:
Can we see if we can make the `key` more unique then this?  Making it the index can potentially cause issues if we removed an item / spliced from this list.  Typically it's not a huge issue, but since this component is generic, we want to make sure if someone does remove an item, it doesn't have a weird side effect with it.